r/src/symbols.cpp

```diff
@@ -75,6 +75,10 @@ SEXP data::classes_arrow_large_list = precious(cpp11::writable::strings(
 SEXP data::classes_arrow_fixed_size_list = precious(cpp11::writable::strings(
     {"arrow_fixed_size_list", "vctrs_list_of", "vctrs_vctr", "list"}));
 
+// TODO: do we even need this?
+SEXP data::classes_arrow_map = precious(
+    cpp11::writable::strings({"arrow_map", "arrow_list", "vctrs_list_of", "vctrs_vctr", "list"}));
+
```

Comment:
I'm a little confused on what this does, and if I need it. Arrow's maps are just lists of structs with some extra validation. What behavior does this affect? How is it tested?

Reply:
@romainfrancois I'm nearly done with this PR, but stumped on this detail. Do you have advice on this?

Reply:
I deleted this part, as the structs don't seem to have this, and maps are made of structs.

Reply:
I think this would be useful for *automatic* creating with `Array$create()` without giving `type=`, the same way as e.g. binary types.

Reply:
Okay. I think I want to leave that for a follow-up. IMO most of the value in implementing map arrays is reading files and streams from other languages; I don't think it's as meaningful a type in R.